### PR TITLE
Use Log4J2

### LINKED ISSUE #25: Logger freezes when called too many times/Use Log4j instead
Steps to reproduce: 
- Download the server pack for ATM4 and place it in the `input` folder
- Open Patchwork's UI interface, and click "Start Patching"
- Observe Patcher freezes while patching mods (after about 5-10 on my machine)
Workaround:
- Comment out all calls to `LOGGER` in `Patchwork.java`
- Observe Patcher now runs as expected and does not freeze

I recommend using Log4j instead, as our custom logger doesn't seem to provide any extra function that cannot be done with Log4j, Log4j will be on the classpath when running Patcher in Minecraft, and having our own logger system is an excessive maintenance burden.

### PR DESCRIPTION
**Pull request by TheGlitch76**
Essentially a copypasta of different stackoverflow solutions cobbled together into ANSI support for Log4J and Swing.
Closes #25 and resolves concerns about when Patcher is run in the game with the upcoming bootstrap mod.

### CODE REVIEW

**Review by Janrupf — APPROVED** [review]

File: src/main/java/com/patchworkmc/ColorPane.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	private static final Color B_Blue = Color.getHSBColor(0.667f, 1.000f, 1.000f);
+	private static final Color B_Magenta = Color.getHSBColor(0.833f, 1.000f, 1.000f);
+	private static final Color B_Green = Color.getHSBColor(0.333f, 1.000f, 1.000f);
+	private static final Color B_Yellow = Color.getHSBColor(0.167f, 1.000f, 1.000f);
+	private static final Color B_Cyan = Color.getHSBColor(0.500f, 1.000f, 1.000f);
+	private static final Color B_White = Color.getHSBColor(0.000f, 0.000f, 1.000f);
+
+	// TODO: instance-based and mutable when we have a dark mode
+	private static final boolean IS_LIGHT = true;
+	private static final Supplier<Color> cReset = () -> IS_LIGHT ? D_Black : D_White;
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/ColorPane.java:33]: Not a huge fan of this. It assumes that dark mode == full black which is often not true.

File: src/main/java/com/patchworkmc/Patchwork.java [hunk 2]
```diff
@@ -168,9 +160,9 @@ private ForgeModJar parseModManifest(Path jarPath) throws IOException, URISyntax
 			toml.load();
 
 			try {
-				accessTransformers = AccessTransformerList.parse(fs.getPath("/META-INF/accesstransformer.cfg"));
+				//accessTransformers = AccessTransformerList.parse(fs.getPath("/META-INF/accesstransformer.cfg"));
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/Patchwork.java]: Uh oh

File: build.gradle [hunk 3]
```diff
@@ -57,11 +57,17 @@ dependencies {
 
 	// Patchwork
 	implementation 'com.patchworkmc:patchwork-manifest:0.1.2'
-	implementation 'com.patchworkmc:application-core:0.1.0'
+	//implementation 'com.patchworkmc:application-core:0.1.0'
```
Comment (coderbot16) [inline build.gradle]: Probably just better to remove the line for now if it's not going to be used

**Review by coderbot16 — CHANGES_REQUESTED** [review]
I'm not a huge fan of log4j2 to be honest, but it seems like it will probably avoid confusing errors for now. There's a few issues I've noticed after looking the code over.

**Review by coderbot16 — APPROVED** [review]

**Pull request merged**